diff --git a/Products/DataCollector/zenmodeler.py b/Products/DataCollector/zenmodeler.py
index 587520910..764860cad 100644
--- a/Products/DataCollector/zenmodeler.py
+++ b/Products/DataCollector/zenmodeler.py
@@ -15,17 +15,7 @@ This information should change much less frequently than performance metrics.
 """
 import Globals
 
-# IMPORTANT! The import of the pysamba.twisted.reactor module should come before
-# any other libraries that might possibly use twisted. This will ensure that
-# the proper WmiReactor is installed before anyone else grabs a reference to
-# the wrong reactor.
-try:
-    import pysamba.twisted.reactor
-    from ZenPacks.zenoss.WindowsMonitor.WMIClient import WMIClient
-    from ZenPacks.zenoss.WindowsMonitor.utils import addNTLMv2Option, setNTLMv2Auth
-    USE_WMI = True
-except ImportError:
-    USE_WMI = False
+USE_WMI = False
 
 import zope.component
 
